packages/kit/src/runtime/server/respond.js

```diff
@@ -392,16 +389,26 @@ export async function respond(request, options, manifest, state) {
 						trailing_slash ?? 'never'
 					);
 				} else if (route.endpoint && (!route.page || is_endpoint_request(event))) {
-					response = await render_endpoint(event, await route.endpoint(), state);
+					const endpoint = await route.endpoint();
+					const response = await render_endpoint(event, endpoint, state);
+
+					// If endpoint doesn't have a handler for these methods, fall back to the page
+					if (
+						response.status === 405 &&
```

Comment:
did you mean for this to be `405`? based on the comment I would have expected `404`

Reply:
Yeah, I think it's 405. Kit responds with `405 method not allowed` when the method handler isn't found.